editor/src/clj/editor/game_object_non_editable.clj

```diff
@@ -47,38 +53,113 @@
         graph (g/node-id->graph-id host-node-id)]
     (g/make-nodes graph [embedded-resource-node-id [embedded-resource-node-type :resource embedded-resource]]
       (project/load-node project embedded-resource-node-id embedded-resource-node-type embedded-resource)
-      (project/connect-if-output embedded-resource-node-type embedded-resource-node-id host-node-id embedded-component-resource-connections))))
+      (project/connect-if-output embedded-resource-node-type embedded-resource-node-id host-node-id embedded-component-connections))))
 
-(g/defnode EmbeddedComponentHostResourceNode
-  (inherits resource-node/ImmutableResourceNode)
+(g/defnk produce-embedded-component-build-targets [embedded-component-build-targets]
+  (into []
+        (map (comp #(assoc % :resource (bt/make-content-hash-build-resource %))
+                   util/only-or-throw))
+        embedded-component-build-targets))
+
+(g/defnk produce-embedded-component-resource-data->scene-index [embedded-component-resource-data->index resource]
+  (let [workspace (resource/workspace resource)
+        ext->resource-type (workspace/get-resource-type-map workspace :non-editable)]
+    (into {}
+          (comp (keep (fn [[embedded-component-resource-data]]
+                        (let [embedded-resource-ext (:type embedded-component-resource-data)
+                              embedded-resource-type (ext->resource-type embedded-resource-ext)
+                              embedded-resource-node-type (:node-type embedded-resource-type)]
+                          (when (g/has-output? embedded-resource-node-type :scene)
+                            embedded-component-resource-data))))
+                (map-indexed (fn [index embedded-component-resource-data]
+                               (pair embedded-component-resource-data index))))
+          (sort-by val embedded-component-resource-data->index))))
+
+(g/defnk produce-referenced-component-proj-path->scene-index [referenced-component-proj-path->index resource]
+  ;; We do a naughty g/node-value inside a production function here, but we only
+  ;; use it to get the Resource associated with the referenced proj-path, so we
+  ;; can obtain the resource-type from it.
```

Comment:
Looks very naughty! Have you considered adding `:scene` output to all resource types that can end up here, and making Script scenes empty?

Reply:
Good point. In the case of a deletion, the `resource-map` won't have an entry for our referenced `proj-path` so we won't find a `node-type`, and that will mess up the indices with this current implementation. I'll need to think a bit about this. 🤔

Reply:
I've updated the code to use a more conventional approach that gets rid of the naughty `g/node-value` call. Nice work catching this!